### ISSUE TITLE
Unknown battery status for several devices

### ISSUE THREAD

**Issue by merlinpimpim** [Koenkk/zigbee2mqtt#24678]
### What happened?

Battery status is unknown for these devices::
- https://www.zigbee2mqtt.io/devices/SNZB-02D.html
- https://www.zigbee2mqtt.io/devices/NAS-AB06B2.html#neo-nas-ab06b2 (Yes, it's been connected for over 24 hours, and the battery level hasn't risen for 6 months).

![image](https://github.com/user-attachments/assets/32d73685-e205-4e09-96df-80cea218bc6c)
(As a control device, the Xiaomi temperature device correctly shows its battery level)

![image](https://github.com/user-attachments/assets/ab773f78-e375-4daa-be34-00d147af5290)

### What did you expect to happen?

Knowing the battery charge level of these devices

### How to reproduce it (minimal and precise)

n/a

### Zigbee2MQTT version

1.41.0

### Adapter firmware version

20240315

### Adapter

SLZB-06

### Setup

HAOS (Green)

**Comment by DaN66O** [comment]
The same problem
https://www.zigbee2mqtt.io/devices/SNZB-01P.html#sonoff-snzb-01p
![Screenshot_1](https://github.com/user-attachments/assets/d43995a1-44f9-4623-84a6-b332c5e52d6c)

**Comment by chris-1243** [comment]
Have you tried to follow the [troubleshoot](https://www.zigbee2mqtt.io/devices/SNZB-02D.html#sonoff-snzb-02d) ?

Mine are perfectly reporting battery value

**Comment by merlinpimpim** [comment]
> Have you tried to follow the [troubleshoot](https://www.zigbee2mqtt.io/devices/SNZB-02D.html#sonoff-snzb-02d) ?
> 
> Mine are perfectly reporting battery value.
> 
> Also check if the `PowerCfg` cluster is bound to the coordinator and if the reporting is correctly set

Yes, (it's the link I posted when I opened the ticket :) )

On the other hand, I forgot to specify that the information does go up in the device details for SNZB-02D:
![image](https://github.com/user-attachments/assets/f37ddf38-226f-4929-8874-1f7c1c6a21ac)
And the PowerCfg cluster seems to be bound correctly.
![image](https://github.com/user-attachments/assets/c091cb5e-7e69-4b0d-9a99-1513aa702d64)

But I think this is another problem with the Neo NAS-AB06B2, because in the Exposes tab, the battery field is “Null”.
![image](https://github.com/user-attachments/assets/8896dcb8-d43c-4d2f-b548-16a71672e00d)

**Comment by chris-1243** [comment]
For your SNZB-02P, it looks more an issue in the frontend. May you try to shirtly press the button on the device to wake it up and press the refresh button close to the battery in the expose tab ?

> But I think this is another problem with the Neo NAS-AB06B2, because in the Exposes tab, the battery field is “Null”.
image.png (view on web)

I had this kind of issue with a Ikea remote. Try to remove the battery, wait some seconds and put it back. Then refresh the value...

**Comment by merlinpimpim** [comment]
> For your SNZB-02P, it looks more an issue in the frontend. May you try to shirtly press the button on the device to wake it up and press the refresh button close to the battery in the expose tab ?
> 
Quickly pressing the device wakes it up (last seen= just now in Z2M), the values are refreshed in Exposes but the question mark remains in the dashboard.
So, yes, probably a fontend problem. Is this the right place to report this type of problem?

> I had this kind of issue with a Ikea remote. Try to remove the battery, wait some seconds and put it back. Then refresh the value...

For the Neo NAS-AB06B2 I tested your proposal (but I'd done it before) without success.
Note that now in Exposes tab the value is N/A instead of Null:
![image](https://github.com/user-attachments/assets/77d8ec44-6544-4058-a1f1-aa12527581c0)

But I think the problem is deeper because in the payload I don't see anything about the battery. Maybe a problem with the converter?
![image](https://github.com/user-attachments/assets/99130c25-c2ae-491c-afc1-3fb20afd35b0)

**Comment by chris-1243** [comment]
Yes, I would bet for the converter and it is a Tuya based sensor... (always tricky).

It would be good to create an external converter and try...I had a quick check and nothing wrong.

Before making a converter, what is it configured in the reporting tab ?

**Comment by merlinpimpim** [comment]
> Before making a converter, what is it configured in the reporting tab ?

Nothing...
![image](https://github.com/user-attachments/assets/0fb5709b-adc2-443b-8e4a-2a8ee8dfae8f)

**Comment by chris-1243** [comment]
Would you try to add this ?

![image](https://github.com/user-attachments/assets/6912770e-6dff-4ef4-9161-d6d52134ba82)

Before clicking on `apply`, awake your sensor. A quick press on the pairing button should do the trick

**Comment by chris-1243** [comment]
> Cache cleared and HA Green rebooted: no change

Frontend problem I guess. I can not help more on that topic.

What clusters do you have on this device ?

**Comment by merlinpimpim** [comment]
> What clusters do you have on this device ?

![image](https://github.com/user-attachments/assets/fb32b0df-1bf8-4e97-bf9f-5d2ddf91fbcb)

**Comment by apydo** [comment]
This is issue is related to #24453.

**Comment by merlinpimpim** [comment]
> This is issue is related to #24453.

Thank you @apydo .
It does indeed look similar for the SNZB-02D, however **I think the problem is different for the Neo NAS-AB06B2** (which has never displayed a battery level in the 1 year I've had it...)


> Would you provide the cluster tab?

![image](https://github.com/user-attachments/assets/6bc6ba14-94a5-4871-b817-81fd3f55e71e)

> Mine are perfectly reporting battery value.

How is this possible @chris-1243  given the https://github.com/Koenkk/zigbee2mqtt/issues/24453 and https://github.com/Koenkk/zigbee2mqtt/issues/24025 tickets?

**Comment by chris-1243** [comment]
> How is this possible @chris-1243 given the #24453 and #24025 tickets?

Firmware version ? I have 4 of them bought at different time and everything is fine (except a randomly polling error which do not have any impact on the reporting of the values)

**Comment by nchevsky** [comment]
🏓

**Comment by merlinpimpim** [comment]
For me it's solved with ZM 2.0.0

**Comment by Sabjul33** [comment]
News for me with the 2.1.0 update !!!  have the battery percentage for the NAS-AB06B2
https://github.com/Koenkk/zigbee2mqtt/issues/19827#issuecomment-2629130678
